src/store/modules/problem_sets.ts

```diff
@@ -43,13 +43,15 @@ export default {
 			_set: ProblemSet): Promise<ProblemSet> {
 			const course_id = rootState.session.course.course_id;
 			// shouldn't we be throwing an error if set_id is null or 0?
-			const response = await api.put(`courses/${course_id}/sets/${_set.set_id ?? 0}`, _set);
+			const response = await api.put(`courses/${course_id}/sets/${_set.set_id ?? 0}`, _set.toObject());
 			const set = response.data as ProblemSet;
-			if (JSON.stringify(set) === JSON.stringify(_set)) {
-				commit('UPDATE_PROBLEM_SET', _set);
-			} else {
-				logger.error(`Problem set #${_set.set_id ?? 0} failed to update properly.`);
-			}
+			// The following is not working.  TODO: fix-me
+			// if (JSON.stringify(set) === JSON.stringify(_set)) {
+			// commit('UPDATE_PROBLEM_SET', _set);
+			// } else {
+			// logger.error(`Problem set #${_set.set_id ?? 0} failed to update properly.`);
+			// }
```

Comment:
The reason this isn't working is because the store holds ProblemSets with keys that start with `_`, but the response from the API does not prepend with `_`.

Reply:
This is a change from the old to new models.  Every time data comes from the server it needs to be parsed and created as a new model. 

Also, this was one place I started testing that the updates were working and there are many places that we need to do this.  Perhaps we can consolidate a lot of code in one location.  

I also intend to develop some unit testing with the server.  That is testing that all of the CRUD commands in the store/modules work.  I think this is a bit tricky with Vuex and maybe the Pinia replacement may be better.